cmd/prysmctl/testnet/generate_genesis.go

```diff
@@ -0,0 +1,246 @@
+package testnet
+
+import (
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"github.com/pkg/errors"
+	fastssz "github.com/prysmaticlabs/fastssz"
+	"github.com/prysmaticlabs/prysm/v3/cmd"
+	"github.com/prysmaticlabs/prysm/v3/config/params"
+	"github.com/prysmaticlabs/prysm/v3/io/file"
+	ethpb "github.com/prysmaticlabs/prysm/v3/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/v3/runtime/interop"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli/v2"
+)
+
+var generateGenesisStateCmd = &cli.Command{
+	Name:   "generate-genesis",
+	Usage:  "Generate a beacon chain genesis state from specified configs",
+	Action: cliActionGenerateGenesisState,
+	Flags: []cli.Flag{
+		cmd.ChainConfigFileFlag,
+		depositJsonFileFlag,
+		configNameFlag,
+		numValidatorsFlag,
+		genesisTimeFlag,
+		outputSSZFlag,
+		outputYamlFlag,
+		outputJsonFlag,
+	},
+}
+
+var (
+	log                 = logrus.WithField("prefix", "genesis")
+	depositJsonFileFlag = &cli.StringFlag{
+		Name:  "deposit-json-file",
+		Value: "",
+		Usage: "Path to deposit_data.json file generated by the staking-deposit-cli tool for optionally specifying validators in genesis state",
+	}
+	configNameFlag = &cli.StringFlag{
+		Name:  "config-name",
+		Usage: "Config kind to be used for generating the genesis state. Default: mainnet. Options include mainnet, interop, minimal, prater, ropsten, sepolia. --chain-config-file will override this flag.",
+		Value: params.MainnetName,
+	}
+	numValidatorsFlag = &cli.Uint64Flag{
+		Name:  "num-validators",
+		Usage: "Number of validators to deterministically generate in the genesis state",
+
+		Value: 0,
+	}
+	genesisTimeFlag = &cli.Uint64Flag{
+		Name:  "genesis-time",
+		Usage: "Unix timestamp seconds used as the genesis time in the genesis state. If unset, defaults to now()",
+	}
+	outputSSZFlag = &cli.StringFlag{
+		Name:  "output-ssz",
+		Usage: "Output filename of the SSZ marshaling of the generated genesis state",
+		Value: "",
```

Comment:
maybe specify a sensible default?

Reply:
These must be empty and at least one must be set by the user. As these are output file paths, I don't want to force a default value

Reply:
I think the idea of these flags is you can get the result in ssz, yaml or json depending on which of the `output-<format>` flags you specify. That said I think that the default value of string flags will be the zero value ("") anyway, so I'd just snip out all these `Value` fields to be more concise.